README.md

```diff
@@ -174,9 +174,93 @@ only works using a custom runner hosted on GCP.**
 The action will automatically detect and use the Application Default
 Credentials.
 
+## Example Workflows
+
+* [Deploy a prebuilt container](#deploy-a-prebuilt-container)
+
+* [Build and deploy a container](#build-and-deploy-a-container)
+
+### Setup
+
+1.  Create a new Google Cloud Project (or select an existing project).
+
+1. [Enable the Cloud Run API](https://console.cloud.google.com/flows/enableapi?apiid=run.googleapis.com).
+
+1.  [Create a Google Cloud service account][sa] or select an existing one.
+
+1.  Add the the following [Cloud IAM roles][roles] to your service account:
+
+    - `Cloud Run Admin` - allows for the creation of new Cloud Run services
+
+    - `Service Account User` -  required to deploy to Cloud Run as service account
+
+    - `Storage Admin` - allow push to Google Container Registry
+
+1.  [Download a JSON service account key][create-key] for the service account.
+
+1.  Add the following [secrets to your repository's secrets][gh-secret]:
+
+    - `GCP_PROJECT`: Google Cloud project ID
+
+    - `GCP_SA_KEY`: the downloaded service account key
+
+### Deploy a prebuilt container
+
+To run this workflow, push to the branch named `example-deploy`:
+
+```sh
+git push YOUR-FORK main:example-deploy
+```
+
+### Build and deploy a container
+
+To run this workflow, push to the branch named `example-build`:
+
+```sh
+git push YOUR-FORK main:example-build
+```
+
+**Reminder: If this is your first deployment of a service, it will reject all unauthenticated requests. Learn more at [allowing unauthenticated requests](#Allow-unauthenticated-requests)**
+
+## Migrating from `setup-gcloud`
+
+Example using `setup-gcloud`:
+
+```YAML
+- name: Setup Cloud SDK
+  uses: google-github-actions/setup-gcloud@v0.2.0
+  with:
+    project_id: ${{ env.PROJECT_ID }}
+    service_account_key: ${{ secrets.GCP_SA_KEY }}
+
+- name: Deploy to Cloud Run
+  run: |-
+    gcloud run deploy $SERVICE \
+      --region $REGION \
+      --image gcr.io/$PROJECT_ID/$SERVICE \
+      --platform managed \
+      --set-env-vars NAME="Hello World" \
+      --allow-unauthenticated
```

Comment:
Should we include the  `--allow-unauthenticated` flag if native action does not set it?

Reply:
sgtm